### DISCUSSION TITLE
Versioning for regex manager based on selected datasource

### DISCUSSION THREAD

**Discussion by thejan2009** [renovatebot/renovate#24292] [Suggest an Idea]
### Tell us more.

I just found a misconfiguration we have with regex manager. The relevant `renovate.json5` excerpt is
``` json
    {
      "fileMatch": [
        "Dockerfile$",
      ],
      "matchStrings": [
        "datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>\\S*)\\s",
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}",
    },

```

With the following Dockerfile snippet using `pypi` datasource, we used the wrong `semver` versioning. To work around the issue, we had to explicitly set `pep440` versioning, although it's already the default for [pypi datasource](https://docs.renovatebot.com/modules/datasource/pypi/).

``` diff
-# renovate: datasource=pypi depName=mkdocs-material
+# renovate: datasource=pypi depName=mkdocs-material versioning=pep440
 ENV MATERIAL_VERSION=9.2.8
 
-# renovate: datasource=pypi depName=mkdocs-multirepo-plugin
+# renovate: datasource=pypi depName=mkdocs-multirepo-plugin versioning=pep440
 ENV MULTIREPO_VERSION=0.6.2
 ```
 To avoid having to go through all the regex definitions in the codebase, I'd propose adding a way to use the selected datasource's default versioning inside `versioningTemplate`.

**Comment by rarkins** [comment]
@secustor do you recall if we currently enforce `versioning` or `versioningTemplate` being mandatory? If so then I think we could relax that so that datasource defaults can be used (which should cover the majority of uses cases and make `versioningTemplate` a rarely used field)

**Reply by secustor** [reply]
I think it is mandatory as regex predates the default versions of datasources.

**Reply by rarkins** [reply]
Feature request created: #24301
